src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,45 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        MachineProducer machineProducer = null;
+        if (type == Bulldozer.class) {
+            machineProducer = new BulldozerProducer();
+            return new ArrayList<>(machineProducer.get());
+        }
+        if (type == Excavator.class) {
+            machineProducer = new ExcavatorProducer();
+            return new ArrayList<>(machineProducer.get());
+        }
+        if (type == Truck.class) {
+            machineProducer = new TruckProducer();
+            return new ArrayList<>(machineProducer.get());
+        }
+        return new ArrayList<>();
```

Comment:
Consider creating a local variable for the list returned by the 'get' method to avoid duplication of 'new ArrayList<>(machineProducer.get())'.